tests/async/tasyncintemplate.nim

```diff
@@ -0,0 +1,12 @@
+discard """
+  output: 42
+"""
+
+import asyncdispatch
+
+template foo() =
+  proc temp(): Future[int] {.async.} = return 42
+  proc tempVoid(): Future[void] {.async.} = echo await temp()
```

Comment:
Can we get a test for a template returning a non-Future to test we get a compile error?

Reply:
=> https://github.com/nim-lang/Nim/pull/17633